datafusion/functions/src/math/power.rs

```diff
@@ -149,22 +151,112 @@ where
 
 /// Binary function to calculate a math power to float exponent
 /// for scaled integer types.
-/// Returns error if exponent is negative or non-integer, or base invalid
 fn pow_decimal_float<T>(base: T, scale: i8, exp: f64) -> Result<T, ArrowError>
 where
     T: From<i32> + ArrowNativeTypeOp,
 {
-    if !exp.is_finite() || exp.trunc() != exp {
+    if exp.is_finite() && exp.trunc() == exp && exp >= 0f64 && exp < u32::MAX as f64 {
+        return pow_decimal_int(base, scale, exp as i64);
+    }
+
+    if !exp.is_finite() {
         return Err(ArrowError::ComputeError(format!(
-            "Cannot use non-integer exp: {exp}"
+            "Cannot use non-finite exp: {exp}"
+        )));
+    }
+
+    pow_decimal_float_fallback(base, scale, exp)
+}
+
+/// Fallback implementation using f64 for negative or non-integer exponents.
+/// This handles cases that cannot be computed using integer arithmetic.
+fn pow_decimal_float_fallback<T>(base: T, scale: i8, exp: f64) -> Result<T, ArrowError>
+where
+    T: From<i32> + ArrowNativeTypeOp,
+{
+    let scale_factor = 10f64.powi(scale as i32);
```

Comment:
There are no new tests so it seems it is not intended but the fallback approach also supports negative scales too, not just exponents.
The integer based approach does not support negative scales (yet) - https://github.com/apache/datafusion/pull/19369/changes#diff-598356e1af0ad23287c968ebf6310366e5127b91a037e076bc264dac3a55e768R125

I'd suggest to either add a check and return error like the integer approach or add unit tests and document it as supported.